### PR TITLE
doc: use pygments devicetree language support

### PR DESCRIPTION
**Pull request by gmarull**
Since Pygments 2.7.0 devicetree syntax is supported, so there is no need to use a custom lexer. 2.9.0 introduces a fix required for Zephyr docs.

### CODE REVIEW

**Review by carlescufi — APPROVED** [review]

**Review by nashif — APPROVED** [review]

**Comment by gmarull** [comment]
@carlescufi @nashif Added an extra commit (2329f0e64cb68ab273d232976d899e25307e4760). I'm not sure if Pygments lexer should actually support fixed cases.

**Comment by carlescufi** [comment]
> @carlescufi @nashif Added an extra commit ([2329f0e](https://github.com/zephyrproject-rtos/zephyr/commit/2329f0e64cb68ab273d232976d899e25307e4760)). I'm not sure if Pygments lexer should actually support fixed cases.

@mbolivar-nordic can review that bit I'm sure.

**Review by mbolivar-nordic — CHANGES_REQUESTED** [review]
I'm not sure about this.

I really like to have the ability to put fragments without the enclosing root node, as is being done in api-usage.rst.

After thinking about it, that's the only blocker I see.

**Comment by gmarull** [comment]
> I'm not sure about this.
> 
> I really like to have the ability to put fragments without the enclosing root node, as is being done in api-usage.rst.
> 
> After thinking about it, that's the only blocker I see.

Actually it only seems to happen when using `a: b@0` syntax, other snippets work fine.

**Comment by mbolivar-nordic** [comment]
> Actually it only seems to happen when using `a: b@0` syntax, other snippets work fine. live editor https://pygments.org/demo/#try

devicetree-spec is using this:

https://github.com/devicetree-org/devicetree-specification/blob/master/source/extensions/DtsLexer.py

It allows this syntax:

Screenshot from 2021-03-17 09-50-48

I don't think we should abandon this ability.

**Comment by mbolivar-nordic** [comment]
Hm, I see the `connector: connector` is missing a unit address. `connector: connector@0` gives me some red highlighting in the demo you posted @gmarull. Nonetheless I feel like upstream should be fixed so we don't have to introduce extraneous syntax in our examples.

**Comment by gmarull** [comment]
FYI https://github.com/pygments/pygments/pull/1755

**Comment by gmarull** [comment]
Necessary changes to support some DT snippets has been fixed upstream, the PR is now ready and waiting for the next release of pygments.

**Comment by mbolivar-nordic** [comment]
> [@carlescufi](/carlescufi) [carlescufi](/carlescufi) requested a review from [mbolivar-nordic](/mbolivar-nordic) [9 hours ago](#event-4555895329)

I'm waiting for the pygments release before approving.

**Comment by gmarull** [comment]
@mbolivar-nordic should be ready now

File: doc/guides/dts/howtos.rst [hunk 1]
```diff
@@ -659,20 +659,22 @@ node with path ``/soc/i2c@12340000`` in a C/C++ file:
 
 And if you're trying to **set** that property in a devicetree overlay:
 
-.. code-block:: DTS
+.. code-block:: devicetree
 
-   /*
-    * foo.overlay: DTS names with special characters, etc.
-    */
+   \{
```
Comment (mbolivar-nordic) [inline doc/guides/dts/howtos.rst:664]: Why is this here? I thought it would not be necessary to artificially put things under the root node anymore

Comment (gmarull) [reply doc/guides/dts/howtos.rst:664]: This is the only case that is not handled. Note that current lexer doesn't seem to process it correctly either:

image

But for some reason it doesn't generate any warning.
https://docs.zephyrproject.org/latest/guides/dts/howtos.html#make-sure-you-re-using-the-right-names

File: scripts/requirements-doc.txt [hunk 2]
```diff
@@ -5,6 +5,7 @@ sphinx~=3.3
 sphinx_rtd_theme>=0.5.2,<1.0
 sphinx-tabs
 sphinxcontrib-svg2pdfconverter
+pygments~=2.9
```
Comment (mbolivar-nordic) [inline scripts/requirements-doc.txt:8]: >=?

Comment (gmarull) [reply scripts/requirements-doc.txt:8]: ~= is the compatible version specifier, means `>=2.9,<3`. I can't find if pygments uses semver though...

File: doc/guides/dts/howtos.rst (hunk 1 above)
Comment (mbolivar-nordic) [reply doc/guides/dts/howtos.rst:664]: How about just `code-block:: none` then?

File: scripts/requirements-doc.txt (hunk 2 above)
Comment (mbolivar-nordic) [reply scripts/requirements-doc.txt:8]: OK, fair

File: doc/guides/dts/howtos.rst (hunk 1 above)
Comment (gmarull) [reply doc/guides/dts/howtos.rst:664]: with current patch:

image

Comment (mbolivar-nordic) [reply doc/guides/dts/howtos.rst:664]: Putting this inside of the root node is just actively wrong for what this example is trying to be about

**Review by mbolivar-nordic — APPROVED** [review]
Thanks for your perseverance! It's nice to be using the upstream lexer.
